src/api/app/controllers/search_controller.rb

```diff
@@ -68,26 +68,45 @@ def attribute
   def missing_owner
     params[:limit] ||= '0' # unlimited by default
 
-    @owners = Owner.search(params, nil).map(&:to_hash)
+    @owners = OwnerSearch::Missing.new(params).find.map(&:to_hash)
+  end
+
+  def owner_group_or_user
+    if params[:user].present?
+      User.find_by_login!(params[:user])
+    elsif params[:group].present?
+      Group.find_by_title!(params[:group])
+    end
+  end
+
+  def owner_package_or_project
+    return if params[:project].blank?
+    if params[:package].present?
+      Package.get_by_project_and_name(params[:project], params[:package])
+    else
+      Project.get_by_name(params[:project])
+    end
   end
 
   def owner
     Backend::Test.start if Rails.env.test?
 
-    obj = nil
-    obj = params[:binary] if params[:binary].present?
-    obj = User.find_by_login!(params[:user]) if params[:user].present?
-    obj = Group.find_by_title!(params[:group]) if params[:group].present?
-    obj = Package.get_by_project_and_name(params[:project], params[:package]) unless params[:project].blank? || params[:package].blank?
-    obj = Project.get_by_name(params[:project]) if obj.nil? && params[:project].present?
+    if params[:binary].present?
+      owners = OwnerSearch::Assignee.new(params).for(params[:binary])
+    elsif (obj = owner_group_or_user)
```

Comment:
no better name than ```obj```? :(

if you are assigning the ```obj``` over and over again, why not just call ```owner_group_or_user``` directly?

i.e

```
if params[:binary].present?
      owners = OwnerSearch::Assignee.new(params).for(params[:binary])
    elsif owner_group_or_user
      owners = OwnerSearch::Owned.new(params).for(owner_group_or_user)
    end
    if owners.nil? && owner_package_or_project
      owners = OwnerSearch::Container.new(params).for(owner_group_or_user)
    end
```
and if possible, you could do some memoization in the method ```owner_group_or_user```

Reply:
why? Doesnt' sound like a good advise to me.

Reply:
no? ok, maybe is just me but this obj thing looks ugly, beside this assignment in the if/elsif statement